File: day3/ContainDuplicate.js
let arr = [1, 2, 3, 4, 1, 5, 2];

function hasDuplicate(nums) {
    let mySet = new Set();
    
    for (let num of nums) {
        if (mySet.has(num)) {
            return true; // Jaise hi pehla duplicate mila, true return kardo
        }
        mySet.add(num);
    }
    return false; // Poora loop khatam hone par bhi kuch nahi mila
}

console.log(hasDuplicate(arr)); // Output: true

File: day3/ContainDuplicate.py
arr = [1, 2, 3, 4, 1, 5, 2]


def has_duplicate(nums):
    my_set = set()

    for num in nums:
        if num in my_set:
            return True  # Jaise hi pehla duplicate mila, true return kardo
        my_set.add(num)

    return False  # Poora loop khatam hone par bhi kuch nahi mila


print(has_duplicate(arr))  # Output: True
